src/serde/de/indexed.rs

```diff
@@ -57,7 +57,7 @@ impl<'de> IndexedDeserializer<'de> {
         IndexedDeserializer {
             source,
             delimiter: if delimiter.len() == 1 {
-                Delimiter::Char(delimiter.chars().nth(0).unwrap())
+                Delimiter::Char(delimiter.chars().next().unwrap())
```

Comment:
Consider using `if let` on `delimiter.chars().next()`, since `delimiter.len()` returns the length in bytes of the String in case that's the active pattern. It also removes the unwrap.

Reply:
Ah, damn unicode xd